cedar-lean/Cedar/Thm/Partial/Evaluation/AndOr.lean

```diff
@@ -152,4 +152,52 @@ theorem subst_preserves_evaluation_to_value {x₁ x₂ : Partial.Expr} {req req'
         case ext x => cases x <;> simp
   }
 
+/--
+  Inductive argument that if partial-evaluation of a `Partial.Expr.and` or
+  `Partial.Expr.or` returns an error, then it also returns an error (not
+  necessarily the same error) after any substitution of unknowns
+
+  The proof of `subst_preserves_evaluation_to_value` for this
+  request/entities/subsmap is passed in as an argument, because this file can't
+  import `Thm/Partial/Evaluation.lean` to access it.
+  Alternately, this entire inductive proof could live in its own set of files,
+  all of which could depend on `Thm/Partial/Evaluation.lean` and its theorems
+  like `subst_preserves_evaluation_to_value`.
```

Comment:
The current design seems fine to me. Thanks for leaving a comment to avoid future confusion. Though I think it would be cleaner to have this "Alternatively..." text in an issue rather than in the code.

Reply:
I'd like to only make the issue if we think we actually want to move to that design.

Reply:
Having thought about it some more, and especially seeing as this pattern will come up a couple more times as I upstream the rest of the PE soundness proof, I think the module-per-toplevel-inductive-proof organization probably makes the most sense.  I'll make the issue.

Reply:
#372

Reply:
Also replaced the "Alternately" text with references to #372.